src/main/kotlin/com.bugsnag.android.gradle/BugsnagPlugin.kt

```diff
@@ -83,28 +105,40 @@ class BugsnagPlugin : Plugin<Project> {
 
             project.afterEvaluate {
                 android.applicationVariants.configureEach { variant ->
-                    registerBugsnagTasksForVariant(project, variant, bugsnag)
+                    registerBugsnagTasksForVariant(
+                        project,
+                        variant,
+                        bugsnag,
+                        httpClientHelperProvider,
+                        releasesUploadClientProvider,
+                        proguardUploadClientProvider,
+                        ndkUploadClientProvider
+                    )
                 }
                 registerNdkLibInstallTask(project, bugsnag, android)
             }
         }
     }
 
-    private fun registerNdkLibInstallTask(project: Project,
-                                          bugsnag: BugsnagPluginExtension,
-                                          android: AppExtension) {
+    private fun registerNdkLibInstallTask(
+        project: Project,
+        bugsnag: BugsnagPluginExtension,
+        android: AppExtension
+    ) {
         val ndkTasks = project.tasks.withType(ExternalNativeBuildTask::class.java)
         val cleanTasks = ndkTasks.filter { it.name.contains(CLEAN_TASK) }.toSet()
         val buildTasks = ndkTasks.filter { !it.name.contains(CLEAN_TASK) }.toSet()
 
         if (buildTasks.isNotEmpty()) {
-            val ndkSetupTask = project.tasks.create("bugsnagInstallJniLibsTask", BugsnagInstallJniLibsTask::class.java)
-            ndkSetupTask.buildDirDestination.set(File(project.buildDir, "/intermediates/bugsnag-libs"))
-            val files = resolveBugsnagArtefacts(project)
-            ndkSetupTask.bugsnagArtefacts.set(files)
+            val ndkSetupTask = BugsnagInstallJniLibsTask.register(project, "bugsnagInstallJniLibsTask") {
+                val files = resolveBugsnagArtifacts(project)
+                bugsnagArtifacts.from(files)
```

Comment:
Just something I wanted to mention - `ConfigurableFileCollection` should conventionally use `from()` rather than `Property.set()`. Fixed these up during this